src/main/java/com/robinpowered/sdk/exception/HttpException.java

```diff
@@ -44,12 +45,18 @@ public static HttpException factory(RetrofitError error) {
         Response response = error != null ? error.getResponse() : null;
 
         if (response != null) {
-            int statusCode = response.getStatus();
+            String responseMessage = null;
+            if (error.getBody() instanceof ApiResponse) {
+                ApiResponse body = (ApiResponse) error.getBody();
+                responseMessage = body.getMeta().getMessage();
```

Comment:
Since this is an error/exception handling mechanism, which could cover a lot of cases where "things went wrong", it may be a good idea to check whether `getMeta()` returns `null` or not before grabbing the message.